submission/submitter/src/main/kotlin/ac/uk/ebi/biostd/submission/model/FilesSource.kt

```diff
@@ -15,8 +14,26 @@ interface FilesSource {
     fun readText(filePath: String): String
 }
 
-class ListFilesSource(private val files: List<ResourceFile>) : FilesSource {
+class MixedFilesSource(
+    private val listFiles: ListFilesSource,
```

Comment:
`ListFilesSource` and `PathFilesSource` names were not the best maybe refactor to FileSystemSource and ProvidedFileSource

Reply:
I was actually thinking about renaming these to `AttachedFilesSource` and `UserFilesSource`. WDYT?